source/adapters/level_zero/v2/queue_api.cpp

```diff
@@ -0,0 +1,321 @@
+/*
+ *
+ * Copyright (C) 2024 Intel Corporation
+ *
+ * Part of the Unified-Runtime Project, under the Apache License v2.0 with LLVM
+ * Exceptions. See LICENSE.TXT
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+ *
+ * @file queue_api.cpp
+ *
+ */
+
+#include "queue_api.hpp"
+
+UR_APIEXPORT ur_result_t UR_APICALL urQueueGetInfo(ur_queue_handle_t hQueue,
```

Comment:
With no need to support the legacy path together with the new one, we might be able to significantly simplify the way we do dispatching, like we originally planned to. But we can revisit that later.